runtime/compiler/optimizer/SequentialStoreSimplifier.cpp

```diff
@@ -1670,8 +1678,15 @@ bool TR_arraycopySequentialStores::insertConsistentTree()
       }
 
    // make sure the index variable and base variable is consistent with the first tree
-   TR::SymbolReference* baseRef = _addrTree[0]->getBaseVarNode()->isNull() ? NULL : _addrTree[0]->getBaseVarNode()->getChild()->skipConversions()->getSymbolReference();
-   if (baseRef != activeBaseRef)
+   TR::Node* baseNode = NULL;
+   TR::SymbolReference* baseRef = NULL;
+   if (!_addrTree[0]->getBaseVarNode()->isNull())
+      {
+      baseNode = _addrTree[0]->getBaseVarNode()->getChild()->skipConversions();
+      if (baseNode->hasSymbolReference())
+         baseRef = baseNode->getSymbolReference();
+      }
+   if (baseRef != activeBaseRef || baseNode != activeBaseNode)
```

Comment:
I think the comparison of `baseRef` and `activeBaseRef` is now redundant.  `baseNode == activeBaseNode` implies `baseRef == activeBaseRef`, while `baseRef != activeBaseRef` implies `baseNode != activeBaseNode`.

Reply:
Make sense, then no reason to get symRefs in the first place.
The other usage of symRef in the code is to compare it to in the value tree.
https://github.com/eclipse-openj9/openj9/blob/73af0c72ea6d61f98bc7b07ff2d6c02128a71d61/runtime/compiler/optimizer/SequentialStoreSimplifier.cpp#L1709-L1714

Which would need to be changed too to verify using nodes instead of symRefs unless it will be too conservative...
A similar recursive tree lookup for nodes would be needed to use instead (if exists?).